File: 2015/solutions/solution_01.js
const fs = require("fs");
const input = fs.readFileSync(__dirname + "/../input/input_01.txt", "utf8")
  .trim();
  
// Part 1
function part1() {
  return input.split("").reduce((l, d) => {
    if (d == "(") return ++l;
    else return --l;
  }, 0);
}

// Part 2
function part2() {
  let l = 0;
  for (let i = 0; i < input.length; i++) {
    if (input[i] == "(") l++;
    else l--;
    if (l < 0) return (i + 1);
  }
}

module.exports = { part1, part2 }

File: 2018/solutions/solution_01.py
with open("./2018/input/input_01.txt") as f:
    lines = f.read().splitlines()

# Part 1
def part1():
    return sum(list(map(int, lines)))

# Part 2
log = set([0])
def part2(sum = 0):
    for shift in lines:
        sum = sum + int(shift)
        if sum in log:
            return sum
        else:
            log.add(sum)
    return part2(sum)
